internal/git/provider/provider_test.go

```diff
@@ -41,7 +41,7 @@ func TestFromURL(t *testing.T) {
 	}
 
 	for _, testCase := range testCases {
-		p, repo, err := provider.FromURL(testCase.url)
+		p, repo, err := provider.FromURL(testCase.url, provider.WithToken("api-token", "token"))
```

Comment:
Why is this necessary?

Reply:
OK; would you put it in a separate commit, and explain in the commit message what was broken and how this fixes it, so future readers understand why this change is here. Thank you!